### PR TITLE
Frontend compatible

### PR DESCRIPTION
**Pull request by LokeSGJ**
# Description

Added activity endpoints and adjusted the design to be compatible with the current frontend design. New endpoints include:

- A GET endpoint to get a single activity by id.
- A GET endpoint to get all activities in a day for a citizen (student).
- A POST endpoint to create an activity.
- A PUT endpoint to update an activity.
- A DELETE endpoint to delete an activity.

Fixes #\<10>

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Has been hand-tested via Swagger (proper integration tests will be added once mocks are set up)

### CODE REVIEW

**Review by MertzAndreas — APPROVED** [review]

File: GirafAPI/Endpoints/ActivityEndpoints.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+using Microsoft.EntityFrameworkCore;
+
+namespace GirafAPI.Endpoints;
+
+public static class ActivityEndpoints
+{
+    public static RouteGroupBuilder MapActivityEndpoints(this WebApplication app)
+    {
+        var group = app.MapGroup("weekplan");
+
```
Comment (Simo205j) [inline GirafAPI/Endpoints/ActivityEndpoints.cs:14]: Kunne denne blive prefixed til "activity" og så slet activity fra get og delete så det er consistent

File: GirafAPI/Endpoints/ActivityEndpoints.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            if (activity is null)
+            {
+                return Results.NotFound();
+            }
+            
+            dbContext.Entry(activity).CurrentValues.SetValues(updatedActivity.ToEntity(id));
+            await dbContext.SaveChangesAsync();
+            
+            return Results.Ok();
+        });
```
Comment (Simo205j) [inline GirafAPI/Endpoints/ActivityEndpoints.cs:58]: Kan endpoint for denne blive gjort consistent så den matcher, GET og DELETE?

**Review by Simo205j — CHANGES_REQUESTED** [review]
Ser fint ud, har lige en enkelt ændring til endpoints for PUT jeg godt kunne tænke mig ændret

File: GirafAPI/Endpoints/ActivityEndpoints.cs (hunk 2 above)
Comment (LokeSGJ) [reply GirafAPI/Endpoints/ActivityEndpoints.cs]: Jo, sagtens. Det er ordnet :)

**Review by MertzAndreas — APPROVED** [review]

**Pull request merged**